### DISCUSSION TITLE
How do I remove the `workspace:` prefix from package.json when running lerna publish in the pnpm workspace?

### DISCUSSION THREAD

**Discussion by imguolao** [lerna/lerna#3338] [Q&A]
I saw the below description on the documents, why？

> During lerna version, dependencies will be updated as normal, but will preserve the workspace: prefix if it exists.

How do I remove the `workspace:` prefix from package.json when running `lerna publish` in the pnpm workspace?

version: lerna@5.5.1

**Comment by zqinmiao** [comment]
The same question!

I used `lerna publish`, Report `lerna ERR! TypeError: Cannot read property 'version' of undefined`，[Related issue](https://github.com/lerna/lerna/issues/3343)

**Comment by ghiscoding** [comment]
[v5.5.0](https://github.com/lerna/lerna/releases/tag/v5.5.0) was supposed to have brought support for that in PR #3284 and if implemented correctly it should take care of replacing the `workspace:` prefix during the publish phase, without any intervention from the user, and it should leave the `workspace:` untouched in every `package.json` (maybe it wasn't phrased properly in their docs, but the process of replacing the `workspace:` with the version to publish should be transparent to the user). If it doesn't do that, then you should definitely open an issue and let them know of the problem or you could alternatively use [Lerna-Lite](https://github.com/ghiscoding/lerna-lite) which works correctly since a few months.

**Comment by zepang** [comment]
I got the smae problem [3393](https://github.com/lerna/lerna/issues/3393)
